src/handlers.py

```diff
@@ -11,23 +11,95 @@
 #     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 #     See the License for the specific language governing permissions and
 #     limitations under the License.
+import base64
 import json
 import logging
+from urlparse import urlparse
+
+from google.appengine.api import users
 
 from base import handlers
+import settings
 
 # Minimal set of handlers to let you display main page with examples
 # class RootHandler(handlers.BaseHandler):
 #
 #   def get(self):
 #     self.render('base.tpl')
 
+
 class CspHandler(handlers.BaseAjaxHandler):
 
-  def post(self):
-    try:
-      report = json.loads(self.request.body)
-      logging.warn('CSP Violation: %s' % (json.dumps(report['csp-report'])))
-      self.render_json({})
-    except:
-      self.render_json({'error': 'invalid CSP report'})
+    def post(self):
+        try:
+            report = json.loads(self.request.body)
+            logging.warn('CSP Violation: %s' % (json.dumps(
+                         report['csp-report'])))
+            self.render_json({})
+        except:
+            self.render_json({'error': 'invalid CSP report'})
+
+
+class MainPage(handlers.BaseHandler):
+    """Serves static index.html with chosen authorization method
+
+    Avaible authorization methods:
+        * basic_auth (default)
+        * google
+
+    Remember:
+        * To choose method edit AUTHORIZATION_METHOD in settings.py
+        * [google_auth] You need to uncomment login: required in
+          app.yaml.base
+        * [google_auth] To update email white list add an email to file:
+          settings.py
+    """
+
+    def check_basic_auth(self):
+        if self.request.authorization:
+            authorization = base64.b64decode(self.request.authorization[1])
+        else:
+            return False
+        login, password = authorization.split(':')
+        return (login == settings.ADMIN_LOGIN and
+                password == settings.ADMIN_PASSWORD)
+
+    def check_google_auth(self):
+        user = users.get_current_user()
+        if not user:
+            return False
+        _, auth_domain = user.email().split("@", 1)
+        is_domain = auth_domain in settings.EMAIL_WHITELIST
+        is_email = user.email() in settings.EMAIL_WHITELIST
+        if is_domain or is_email:
+            return True
```

Comment:
You should explicitly `return False` at the end of the function, None is falsey but `import this`